### PR TITLE
Separate types into interface files, generate typings from interfaces, switch to jest, update other dependencies

### PR DESCRIPTION
**Pull request by cdashrowe**
…files for build, update package.json to use new generated file

### CODE REVIEW

**Comment by elmarti** [comment]
image

**Comment by elmarti** [comment]
Just one thing, if/when you go and clean up the branches, that `article` branch is referenced in the article https://medium.com/pebblepad/how-to-roll-your-own-keyboard-shortcuts-and-sleep-at-night-a3ad0cd2f0bf

**Comment by elmarti** [comment]
Travis build is failing, likely this? https://github.com/facebook/jest/issues/8218


 TLDR: 
`tsconfig.json` needs `"moduleResolution": "node"`

**Comment by cdashrowe** [comment]
> Travis build is failing, likely this? [facebook/jest#8218](https://github.com/facebook/jest/issues/8218)
> 
> TLDR:
> `tsconfig.json` needs `"moduleResolution": "node"`

Perfect, even if a little odd. 🤷‍♂️

**Comment by cdashrowe** [comment]
Needs version bump.

**Comment by elmarti** [comment]
FYI, it looks like it's associated with my NPM profile and people are actively using it, I could've sworn we did something to set up organisational access, I definitely wouldn't have gone cowboy on it :jerpspin: I will await your command.

**Comment by cdashrowe** [comment]
> FYI, it looks like it's associated with my NPM profile and people are actively using it, I could've sworn we did something to set up organisational access, I definitely wouldn't have gone cowboy on it :jerpspin: I will await your command.

It is listed as an organisational package, maybe I can take ownership of it, I am unsure. 

Will look into it.

Edit. Had a look on npm, it seems yourself and I are listed as maintainers, so access should be okay.

**Review by Exigerr — APPROVED** [review]

**Review by JodiePebble — APPROVED** [review]
